### ISSUE TITLE
Store module registered in `fetch` is not visible in vuex instance when `fetch` is called client side

### ISSUE THREAD

**Issue by przemkow** [nuxt/nuxt.js#7046]
### Version

[v2.11.0](https://github.com/nuxt.js/releases/tag/v2.11.0)

### Reproduction link

[https://codesandbox.io/s/jovial-cannon-iirhd](https://codesandbox.io/s/jovial-cannon-iirhd)

### Steps to reproduce

1. Open Reproduction link
2. Click on "Go to page with dynamic module"
3. Check console.log

### What is expected ?

State of the module registered dynamically is available on the vuex store instance.
console.log message:
//  dynamic module state: {                                    
//    foo: 'bar'
//  }

### What is actually happening?

State of the module registered dynamically is not available on the vuex instance.
console.log message:
//dynamic module state: undefined

### Additional comments?

This behaviour is visible only when `fetch` is called on client-side. When the user refreshes the page and `fetch` is called on server-side everything works as expected.

<div align="right"><sub><em>This bug report is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt.js/issues/c10349">#c10349</a>)</em></sub></div>

**Comment by przemkow** [comment]
**TLDR;(Fix):**
Explicitly add `{ preserveState: false }` to `registerModule` function. ex. 
```
fetch(context) {
  context.store.registerModule(storeNamespace, store, { preserveState: false })
}
```

**Explanation:**
I'm not so familiar with Nuxt.js codebase but it seems that this problem might be caused by that line:
https://github.com/nuxt/nuxt.js/blob/a3fdba885e43a76ad605f21d0cfe95038b2681a2/packages/vue-app/template/index.js#L76-L80

In that line, it seems that Nuxt.js overwrites default behaviour of `store.registerModule` method. Because of that by default, any call to `registerModule()` will contain also 3rd argument - `{ preserveState: true/false}` depending on current environment (client-side - true, server-side - false). In that case, if you are trying to register a new dynamic module inside `fetch` and the user navigates from page A to page B `fetch` will be called on client-side which means that new module will try to preserve existing state which results in not adding dynamic module's state to the store (which is correct [vuex behaviour](https://vuex.vuejs.org/guide/modules.html#preserving-state))

PS. In most of the cases, it's probably a better idea to register store module inside `plugins` (like that store hydration will work out of the box) however in our case it's a bit more complex so we wanted to keep that logic at the "Page" level 🙂

**Comment by Aareksio** [comment]
Hi,

I'd like to reopen the issue, as it is still present and reproducible with the reproduction provided by OP. I just spent a few hours digging into why my dynamic module doesn't register state properly, and what OP wrote is exactly my conclusion.

I'm loading reusable dynamic module via `this.$store.registerModule` when an user opens modal, in `created` lifecycle hook of the modal component. To my knowledge this use-case can not be achieved without passing `{ preserveState: false }`.

Like OP I'm not familiar with Nuxt codebase, if easy fix is not possible, please consider documenting the problem and solution :)
